iam_pipes.tf

```diff
@@ -0,0 +1,396 @@
+        "ec2:DescribeVpcs",
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ]
+    }
+
+    msk = {
+      # Read this for more: https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-pipes-msk.html#pipes-msk-permissions-iam-policy
+      actions = [
+        "kafka:DescribeClusterV2",
+        "kafka:GetBootstrapBrokers",
+        "ec2:CreateNetworkInterface",
+        "ec2:DeleteNetworkInterface",
+        "ec2:DescribeNetworkInterfaces",
+        "ec2:DescribeSecurityGroups",
+        "ec2:DescribeSubnets",
+        "ec2:DescribeVpcs",
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ]
+    }
+
+    lambda = {
+      actions = [
+        "lambda:InvokeFunction"
+      ]
+    }
+
+    step_functions = {
+      actions = [
+        "states:StartExecution"
+      ]
+    }
+
+    api_gateway = {
+      actions = [
+        "execute-api:Invoke"
+      ]
+    }
+
+    api_destination = {
+      actions = [
+        "events:InvokeApiDestination"
+      ]
+    }
+
+    batch = {
+      actions = [
+        "batch:SubmitJob"
+      ]
+    }
+
+    logs = {
+      actions = [
+        "logs:DescribeLogGroups",
+        "logs:DescribeLogStreams",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ]
+    }
+
+    ecs = {
+      actions = [
+        "ecs:RunTask"
+      ]
+    }
+
+    ecs_iam_passrole = {
+      actions = [
+        "iam:PassRole"
+      ]
+    }
+
+    eventbridge = {
+      actions = [
+        "events:PutEvents"
+      ]
+    }
+
+    firehose = {
+      actions = [
+        "firehose:PutRecord"
+      ]
+    }
+
+    inspector = {
+      actions = [
+        "inspector:CreateAssessmentTemplate" # ???
+      ]
+    }
+
+    redshift = {
+      actions = [
+        "redshift-data:ExecuteStatement" # ???
+      ]
+    }
+
+    sagemaker = {
+      actions = [
+        "sagemaker:CreatePipeline" # ???
+      ]
+    }
+
+    sns = {
+      actions = [
+        "sns:Publish"
+      ]
+    }
+  }
+}
+
+#################################
+# IAM role for EventBridge Pipes
+#################################
+
+data "aws_iam_policy_document" "assume_role_pipe" {
+  for_each = local.role_for_pipes
+
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["pipes.amazonaws.com"]
+    }
+
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceAccount"
+      values   = [data.aws_caller_identity.current.account_id]
+    }
+
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceArn"
+      values   = ["arn:aws:pipes:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:pipe/${each.value.Name}"]
+    }
+  }
+}
+
+resource "aws_iam_role" "eventbridge_pipe" {
+  for_each = local.role_for_pipes
+
+  name                  = each.value.role_name
```

Comment:
Should we have support for `name_prefix` and use that as the default? If someone stamps out multiple implementations of this module, what are the chances of role name collisions?

Reply:
Good point, fixed.